packages/api/test/e2e/promise-tx.spec.ts

```diff
@@ -178,21 +183,22 @@ describe.skip('e2e transactions', () => {
   it('makes a transfer with ERA (signAndSend) with invalid time', async (done) => {
     const nonce = await api.query.system.accountNonce(keyring.alice.address());
     const signedBlock = await api.rpc.chain.getBlock();
-    const currentHeight = signedBlock.block.header.number;
+    const currentHeight = (signedBlock as SignedBlock).block.header.number;
     const exERA = new ExtrinsicEra({ current: currentHeight, period: 4 });
     const eraBirth = exERA.asMortalEra.birth(currentHeight.toNumber());
     const eraDeath = exERA.asMortalEra.death(currentHeight.toNumber());
-    console.log('STARTED AT :'+eraBirth+' EXPIRED AT :'+eraDeath);
+    console.log('STARTED AT :' + eraBirth + ' EXPIRED AT :' + eraDeath);
     const eraHash = await api.rpc.chain.getBlockHash(eraBirth);
     const ex = api.tx.balances.transfer(keyring.eve.address(), 12345);
-    const unsubscribe = await api.rpc.chain.subscribeNewHead(async(header) => {
-      console.log(`Chain is at block: #${header.blockNumber}`);
-      if (header.blockNumber.toNumber() === eraDeath-1) {
-        const hash = await ex.signAndSend(keyring.alice, {blockHash: eraHash, era:exERA, nonce});
+
+    const unsubscribe = await api.rpc.chain.subscribeNewHead(async (header) => {
+      if (header.blockNumber.toNumber() === eraDeath - 1) {
+        const hash = await ex.signAndSend(keyring.alice, { blockHash: eraHash, era: exERA, nonce });
 
         expect(hash).toBeUndefined();
         done();
       }
+      unsubscribe();
```

Comment:
Should probably unsubscribe before the done

Reply:
I just removed it completely.